config/pmd.xml

```diff
@@ -15,11 +15,12 @@
     <exclude name="ModifiedCyclomaticComplexity"/>
     <!-- we are using CyclomaticComplexity -->
     <exclude name="StdCyclomaticComplexity"/>
+    <!-- It produces too much violations, suppressed till we figure out how useful that metrics -->
+    <exclude name="NcssCount"/>
```

Comment:
Ok for migration time, but please make a separate issue on this to review. We use same metric as Check, and it works.

Please use issue link as reason of suppression.

Reply:
Removing the suppression `NsccCount` lead to  645 PMD violations like this:
```
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.utils.CommonUtilsTest:147 Rule:NcssCount Priority:3 The method 'testFileExtensions()' has a NCSS line count of 12..
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.utils.FilterUtilsTest:69 Rule:NcssCount Priority:3 The method 'testExceptionOnClosing()' has a NCSS line count of 12..
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.utils.JavadocUtilsTest:108 Rule:NcssCount Priority:3 The method 'testTagPositions()' has a NCSS line count of 13..
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.utils.JavadocUtilsTest:167 Rule:NcssCount Priority:3 The method 'testEmptyBlockCommentAst()' has a NCSS line count of 13..
```
The rule has no property to configure the minimal amount of lines of the code per method.
It is useless for now.

Reply:
> NcssCount 
> The rule has no property to configure the minimal amount of lines of the code per method.

https://pmd.github.io/pmd-6.0.1/pmd_rules_java_design.html#ncsscount
What about `methodReportLevel`, which has a default value of 12?

Reply:
please move this separate issue.
please suppress certain cases(not whole rule) for now till new issue.
I need to review is it Pmd problems of calculation or on our side, as we have the same metric implementation.

Reply:
>please suppress certain cases(not whole rule) for now till new issue.

@romani Are you seriously? There is 645 "certain cases". You want to suppress every class?
The output of this rule is here
https://gist.github.com/pbludov/0a62abdce8853aa8e41b85f1687acb2c

There is 154 classes that violates this rule in the folder `src/main` and 63 classes in the folder `src/test`.

Reply:
My bad, missed this in your comment, for some reason thought there was only 4 cases.
Ok to suppress whole rule.
We use it is our default - https://github.com/checkstyle/checkstyle/blob/master/config/checkstyle_checks.xml#L403
Default limits are http://checkstyle.sourceforge.net/config_metrics.html#JavaNCSS

Pmd rule default are different https://pmd.github.io/pmd-6.0.1/pmd_rules_java_design.html#ncsscount
Can we configure pmd rule the same as our ? 
If possible lets do this, if not reason of suppression should be different. If too hard move it separate issue.

Reply:
Done. Setting `methodReportLevel` to 32 reduces the list to a HUGE but, at least, maintainable amount.